src/framework/audio/internal/rpc/queuedrpcchannel.h

```diff
@@ -0,0 +1,68 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2021 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_QUEUEDRPCCHANNEL_H
+#define MU_AUDIO_QUEUEDRPCCHANNEL_H
+
+#include <thread>
+#include <mutex>
+#include <queue>
+
+#include "irpcchannel.h"
+#include "invoker.h"
+
+namespace mu::audio::rpc {
+class QueuedRpcChannel : public IRpcChannel
+{
+public:
+    QueuedRpcChannel() = default;
+
+    bool isSerialized() const override;
+
+    void send(const Msg& msg) override;
+    ListenID listen(Handler h) override;
+    void unlisten(ListenID id) override;
+
+    bool isWorkerThread() const;
+    void setupWorkerThread(); //! NOTE Must called from worker thread
+
+    void setupMainThread(); //! NOTE Must called from main thread
+
+    void process();
+
+private:
+
+    using MQ = std::queue<Msg>;
```

Comment:
Better to rename to MessageQueue

Reply:
I would prefer to leave it this way, it is a private type, used only in a couple of places inside the internal implementation of the class.